src/main/java/bio/terra/workspace/service/resource/controlled/flight/delete/notebook/RetrieveNotebookServiceAccountStep.java

```diff
@@ -0,0 +1,68 @@
+package bio.terra.workspace.service.resource.controlled.flight.delete.notebook;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.DELETE_NOTEBOOK_SERVICE_ACCOUNT_EMAIL;
+
+import bio.terra.cloudres.google.notebooks.AIPlatformNotebooksCow;
+import bio.terra.cloudres.google.notebooks.InstanceName;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledAiNotebookInstanceResource;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import com.google.api.services.notebooks.v1.model.Instance;
+import java.io.IOException;
+
+/**
+ * A step to retrieve the service account used on a AI Platform Notebook instance before deleting
+ * the instance.
+ *
+ * <p>We don't store the service account email anywhere in Workspace Manager metadata, so we have to
```

Comment:
I don't think so. We don't need to duplicate state that's available in GCP. Even though it doesn't change now, it's a future opportunity to mess it up. And since we don't need to yet, YAGNI.